infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -78,49 +112,54 @@ private void keepAlive(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r)
         r.sendResponse(ctx, msg, CorfuMsgType.ACK.msg());
     }
 
+    //TODO(Zach): Do we need a KEEP_ALIVE?
+
     /**
      * Respond to a version request message.
      *
-     * @param msg   The incoming message
+     * @param req   The incoming message
      * @param ctx   The channel context
      * @param r     The server router.
      */
-    @ServerHandler(type = CorfuMsgType.VERSION_REQUEST)
-    private void getVersion(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        VersionInfo vi = new VersionInfo(serverContext.getServerConfig(),
-                                         serverContext.getNodeIdBase64());
-        r.sendResponse(ctx, msg, new JSONPayloadMsg<>(vi, CorfuMsgType.VERSION_RESPONSE));
+    @RequestHandler(type = RequestPayloadMsg.PayloadCase.VERSION_REQUEST)
+    private void getVersion(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        VersionInfo vi = new VersionInfo(serverContext.getServerConfig(), serverContext.getNodeIdBase64());
```

Comment:
`vi` can be improved. Maybe just use `versionInfo`

Reply:
Done.